addons/mail/static/src/components/notification_alert/notification_alert.js

```diff
@@ -1,14 +1,23 @@
 /** @odoo-module **/
 
-import { registerMessagingComponent } from '@mail/utils/messaging_component';
+import { registry } from "@web/core/registry";
 
 const { Component } = owl;
 
-export class NotificationAlert extends Component {}
+export class NotificationAlert extends Component {
+
+    setup() {
+        super.setup();
+        this.env.services.messaging.modelManager.messagingCreatedPromise.then(() => {
+            this.messaging = this.env.services.messaging.modelManager.messaging;
+            this.render();
+        });
+    }
```

Comment:
It's best to follow the container pattern, like what we did with `ChatterContainer`. So just an OWL component that deploys a messaging component.

That way it's much easier to maintain this business code. Even now, it doesn't properly observe `Messaging/isNotificationBlocked`, so the rendering can show false and outdated info.

Reply:
I have updated the PR :)